### PR TITLE
GAWB-1144: show estimated egress charges in file download window

### PR DESCRIPTION
**Pull request by MatthewBemis**
related: https://github.com/broadinstitute/firecloud-orchestration/pull/321

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/common.cljs [hunk 1]
```diff
@@ -131,6 +131,11 @@
               (loop (/ b 1000) (inc n))))]
     (loop bytes 0)))
 
+(defn format-price [amount]
```
Comment (breilly2) [inline src/cljs/org/broadinstitute/firecloud_ui/common.cljs:134]: I like this. I should switch my cost display to use it.

File: src/cljs/org/broadinstitute/firecloud_ui/common/gcs_file_preview.cljs [hunk 2]
```diff
@@ -18,12 +17,13 @@
       {:header "File Details"
        :content
        (let [{:keys [data error status]} (:response @state)
-             data-size (when data (data "size"))
+             data-size (:size data)
```
Comment (breilly2) [inline src/cljs/org/broadinstitute/firecloud_ui/common/gcs_file_preview.cljs:20]: I'm not familiar enough with this yet, but is this safe to do if `data` is falsy? The previous code seemed to be trying to guard against that.

Comment (MattPutnam) [reply src/cljs/org/broadinstitute/firecloud_ui/common/gcs_file_preview.cljs:20]: Yes, if `data` is `nil`, then `(:size data)` will be `nil`.  One of the perks of working with keyword-keyed maps.

**Comment by breilly2** [comment]
👍

**Review by dvoet — APPROVED** [review]
👍

**Pull request merged**